src/Socket.php

```diff
@@ -18,4 +18,11 @@ public function read(?Cancellation $cancellation = null, ?int $limit = null): ?s
     public function getLocalAddress(): SocketAddress;
 
     public function getRemoteAddress(): SocketAddress;
+
+    public function getTlsState(): TlsState;
```

Comment:
Should this really be part of the basic socket interface? We should probably have a look at the places where it's used.

Reply:
I was debating between putting it in `Socket` and introducing another interface. If we have a mock socket or an implementation that doesn't support TLS, the implementation of these methods will be very simple. Thus I chose to put the methods directly on `Socket`.

I'm not hard set on this decision. Are there circumstances where it be harmful for these methods to be on `Socket`?

Reply:
Oh, my question was rather: Should it be on Socket or EncryptableSocket? Should code only having access to the socket type call this method?

Reply:
Oh you meant specifically `getTlsState()`. This probably should stay only on `EncryptableSocket`. `getTlsInfo()` returning null or not is enough on `Socket` I think.

Reply:
Moved `getTlsState()` back to `EncyptableSocket`.